### PR TITLE
alertmanager boot via go

### PR DESCRIPTION
**Pull request by lanej**

### CODE REVIEW

File: alertmanager/rootfs/boot.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
+	c, err := client.New(cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	kapi := client.NewKeysAPI(c)
+
+	ctx, _ := context.WithTimeout(context.TODO(), 100*time.Millisecond)
```
Comment (arschles) [inline alertmanager/rootfs/boot.go:45]: @lanej better to use `context.Background()` here unless there's a good reason to use TODO instead. also, the 2nd parameter is a [`CancelFunc`](https://godoc.org/golang.org/x/net/context#CancelFunc) which you'll probably want to `defer cancel()` as well

File: alertmanager/rootfs/boot.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	etcdErr := errors.New("waiting for etcd")
```
Comment (arschles) [inline alertmanager/rootfs/boot.go:47]: can this go into a `var` outside of main?

File: alertmanager/rootfs/boot.go (hunk 1 above)
Comment (lanej) [inline alertmanager/rootfs/boot.go:45]: @arschles as i am brand spanking new to go, this is very helpful thanks.  still trying to wrap my head around the finer points of the `net/http` lib

Comment (arschles) [inline alertmanager/rootfs/boot.go:45]: @lanej you're welcome. I'm open for questions whenever. Also, `context.Context` is not an official part of `net/http`, but it's very helpful.

**Comment by lanej** [comment]
confd is a monster.  not enough benefit to writing in shell. outie.

**Pull request closed without merging**